source/model.js

```diff
@@ -4,103 +4,285 @@
 
 /********************** Model **********************/
 "use strict";
+let playerTypes = {
+    'human' : 'Player',
+    'computer' : 'ComputerPlayer'
+};
 
-let singletonField = (function(){
+let singletonGameDataManager = (function(){
     let instance;
 
-    function createInstance(){
-        return new Field();
+    function GameDataManager() {
+        let players = [
+            {type: 'human', name: 'Player', symbol: 'X'},
+            {type: 'computer', name: 'Computer', symbol: 'O'}
+        ];
+
+        let statusList = ['Playing', 'Winner', 'Tie'];
+
+        let _rows = 5;
+        let _columns = 4;
+        let _winLength = 3;
+
+        this.getStatusList = function(){
+            return statusList;
+        };
+
+        this.getRows = function(){
+            return _rows;
+        };
+
+        this.getColumns = function(){
+            return _columns;
+        };
+
+        this.getWinLength = function(){
+            return _winLength;
+        };
+
+        this.setRows = function(rows){
+            if (Number.isInteger(rows) && rows > 2 && rows <= 10){
+                _rows = rows;
+                validateWinLength(rows);
+            }
+        };
+
+        this.setColumns = function(columns){
+            if (Number.isInteger(columns) && columns > 2 && columns <= 10) {
+                _columns = columns;
+                validateWinLength(columns);
+            }
+        };
+
+        this.setWinLength = function(winLength){
+            if (Number.isInteger(winLength) && winLength > 2 && winLength <= getDiagonal()) {
+                _winLength = winLength;
+            }
+        };
+
+        this.getPlayerList = function(){
+            return players.slice(0);
+        };
+
+        this.updatePlayer = function(name, prop, value){
+            if (prop !== 'type') {
+                players.find(function(item) {
+                    return item.name === name;
+                })[prop] = value;
+            }
+        };
+
+        this.addPlayer = function(type, name, symbol){
+            type = type.toLowerCase();
+            if (canAddPlayer() && playerTypes[type] && playerTypes[type] !== undefined) {
+                for (let i = 0; i < players.length; i++) {
+                    if (players[i].name === name)       name = name + '' + i;
+                    if (players[i].symbol === symbol)   symbol = symbol + '' + i;
+                }
+                let newPlayer = {
+                    type: type,
+                    name: name,
+                    symbol: symbol
+                };
+                players.splice(players.length, 0, newPlayer);
+            }
+            else console.log('Cannot add player');
```

Comment:
Please do not leave console.log or any other debug-centric code in the codebase.